src/protocol/payload/block.rs

```diff
@@ -174,17 +183,15 @@ impl Codec for Header {
 
 impl Header {
     /// Calculates the double Sha256 hash for [Header]
-    pub fn double_sha256(&self) -> std::io::Result<Hash> {
+    pub fn double_sha256(&self) -> Hash {
         let mut buffer = Vec::new();
 
-        self.encode_without_tx_count(&mut buffer)?;
+        self.encode_without_tx_count(&mut buffer).unwrap();
```

Comment:
I'm not convinced we'd want to unwrap this? In general having protocol implementation code return errors would be better, we can always unwrap at the test level.

Reply:
Reverted